docs/proposals/20210713-sidecarsetimagepullsecrects.md

```diff
@@ -0,0 +1,105 @@
+---
+title: SidecarSet ImagePullSecrets
+
+authors:
+  - "@veophi"
+
+reviewers:
+  - "@Fei-Guo"
+  - "@furykerry"
+  - "@FillZpp"
+
+creation-date: 2021-07-13
+last-updated: 2021-07-13
+status: implementable
+---
+
+# SidecarSet ImagePullSecrets
+- Provide a way to pull images using `Secret` from private repositories for `SidecarSet`.
+
+## Table of Contents
+A table of contents is helpful for quickly jumping to sections of a proposal and for highlighting
+any additional information provided beyond the standard proposal template.
+[Tools for generating](https://github.com/ekalinin/github-markdown-toc) a table of contents from markdown are available.
+
+- [SidecarSet ImagePullSecrets](#sidecarset-imagepullsecrets)
+  - [Table of Contents](#table-of-contents)
+  - [Motivation](#motivation)
+  - [Proposal](#proposal)
+    - [User Story](#user-story)
+    - [Implementation Plan](#implementation-plan)
+  - [Implementation History](#implementation-history)
+
+## Motivation
+In k8s, `Secret` is an easy and safe way to store and manage sensitive information, such as passwords, OAuth tokens, and ssh keys.
+One of the most common-used features of it is to pull images from private repositories. However, `SidecarSet` does not support this feature so far.
+
+## Proposal
+**Main idea**: In this design, we separate the logic of `Secret` and `SidecarSet`.
+In `SidecarSet` part, we only consider injecting their `imagePullSecrets` feilds into Pod.
+Users should manually  distribute the required `Secrets` to all the namespaces that the `SidecarSet` may be instantiated.
+
+### API Definition
+We add `imagePullSecrets` field in `apis/apps/v1alpha1/sidecarset_types.go`:
+```go
+type SidecarSetSpec struct {
+	...
+	// List of the names of secrets required by pulling sidecar container images
+	ImagePullSecrets []corev1.LocalObjectReference `json:"imagePullSecrets,omitempty"`
```

Comment:
`corev1.LocalObjectReference` just has a field `name`, where is the namespace of the secret?